src/osd/OSD.cc

```diff
@@ -1958,6 +1947,10 @@ int OSD::init()
   if (is_stopping())
     return 0;
 
+  service.init();
+  service.publish_map(osdmap);
+  service.publish_superblock(superblock);
```

Comment:
Yeah.. my test was hanging because we

 - start objecter, which requests an osdmap
 - osd does more authentication stuff
 - mon sends osdmap
 - osd ignores osdmap since init isn't done
 - osd finish init, starts boot.. but doesn't have latest map.  requests it, but mon doesn't send
 - osd thinks we're still on the old map, which says firefly osds are booted with incompat features, and does nothing.

This bug has been there forever but doesn't usually matter because the osd doesn't usually stop itself from trying to boot.. and because that doesn't usually happen right when it starts up.  (Fixed this separately in the bigbang testing a couple weeks back).